File: xmdebug/src/main/java/com/leo/xmdebug/DebugHomeActivity.java
package com.leo.xmdebug;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.leo.baseui.mutiType.base.Items;
import com.leo.xmdebug.home.DebugSwitchMultiProvider;
import com.leo.xmdebug.home.model.DebugLocalDataMultiModel;
import com.leo.xmdebug.home.model.DebugLogMultiModel;
import com.leo.xmdebug.home.model.DebugSwitchMultiModel;
import com.leo.xmdebug.home.provider.DebugLocalDataMultiProvider;
import com.leo.xmdebug.home.provider.DebugLogMultiProvider;
import com.leo.xmdebug.main.DebugTopMultiProvider;
import com.leo.xmdebug.main.adapter.DebugListAdapter;
import com.leo.xmdebug.main.model.DebugTopMultiModel;
import com.leo.xmdebug.widget.floating.model.DebugBaseInfoMultiModel;
import com.leo.xmdebug.widget.floating.provider.DebugBaseInfoMultiProvider;

public class DebugHomeActivity extends DebugBaseActivity {
    private RecyclerView contentRv;
    private DebugListAdapter adapter;

    public DebugHomeActivity() {
    }

    public static void enterActivity(Context context) {
        Intent intent = new Intent(context, DebugHomeActivity.class);
        context.startActivity(intent);
    }

    protected int getLayoutId() {
        return R.layout.activity_home;
    }

    protected void init() {
        this.initToolBar();
        this.initViews();
        this.initData();
    }

    private void initToolBar() {
        this.getToolbar().setTitle("Debug");
    }

    private void initViews() {
        this.contentRv = this.findViewById(R.id.content_rv);
    }

    private void initData() {
        this.adapter = new DebugListAdapter();
        this.adapter.register(DebugTopMultiModel.class, new DebugTopMultiProvider(this));
        this.adapter.register(DebugBaseInfoMultiModel.class, new DebugBaseInfoMultiProvider(this));
        this.adapter.register(DebugSwitchMultiModel.class, new DebugSwitchMultiProvider(this));
        this.adapter.register(DebugLocalDataMultiModel.class, new DebugLocalDataMultiProvider(this));
        this.adapter.register(DebugLogMultiModel.class, new DebugLogMultiProvider(this));
        this.contentRv.setLayoutManager(new LinearLayoutManager(this));
        this.contentRv.setAdapter(this.adapter);
        Items items = new Items();
        items.add(new DebugTopMultiModel());
        items.add(new DebugBaseInfoMultiModel());
        items.add(new DebugSwitchMultiModel());
        items.add(new DebugLocalDataMultiModel());
        items.add(new DebugLogMultiModel());
        this.adapter.setItems(items);
    }
}

File: xmdebug/src/main/java/com/leo/xmdebug/DebugHomeActivity.kt
package com.leo.xmdebug

import android.content.Context
import android.content.Intent
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.leo.baseui.mutiType.base.Items
import com.leo.baseui.ui.BaseActivity
import com.leo.litang.db.TestAccessor
import com.leo.xmdebug.home.DebugSwitchMultiProvider
import com.leo.xmdebug.home.model.DebugLocalDataMultiModel
import com.leo.xmdebug.home.model.DebugLogMultiModel
import com.leo.xmdebug.home.model.DebugSwitchMultiModel
import com.leo.xmdebug.home.provider.DebugLocalDataMultiProvider
import com.leo.xmdebug.home.provider.DebugLogMultiProvider
import com.leo.xmdebug.main.adapter.DebugListAdapter
import com.leo.xmdebug.utils.CommonStatusPrefManager
import com.leo.xmdebug.widget.floating.model.DebugBaseInfoMultiModel
import com.leo.xmdebug.widget.floating.provider.DebugBaseInfoMultiProvider


class DebugHomeActivity : BaseActivity() {
    private lateinit var contentRv: RecyclerView
    private lateinit var adapter: DebugListAdapter

    override fun getLayoutId(): Int {
        return R.layout.activity_home
    }

    override fun init() {
        this.initToolBar()
        this.initViews()
        this.initData()
    }

    private fun initToolBar() {
        this.toolbar.setMiddleTitle("Debug")

    }

    private fun initViews() {
        this.contentRv = this.findViewById(R.id.content_rv)
    }

    private fun initData() {
        saveData()


        this.adapter = DebugListAdapter()
        this.adapter.register(DebugBaseInfoMultiModel::class.java, DebugBaseInfoMultiProvider(this))
        this.adapter.register(DebugSwitchMultiModel::class.java, DebugSwitchMultiProvider(this))
        this.adapter.register(DebugLocalDataMultiModel::class.java, DebugLocalDataMultiProvider(this))
        this.adapter.register(DebugLogMultiModel::class.java, DebugLogMultiProvider(this))
        this.contentRv.layoutManager = LinearLayoutManager(this)
        this.contentRv.adapter = this.adapter
        val items = Items()
        items.add(DebugBaseInfoMultiModel())
        items.add(DebugSwitchMultiModel())
        items.add(DebugLocalDataMultiModel())
        items.add(DebugLogMultiModel())
        this.adapter.items = items
    }

    private fun saveData() {
        CommonStatusPrefManager.saveOpenTime(this)
        TestAccessor(this).insert()
    }

    companion object {

        @JvmStatic
        fun enterActivity(context: Context) {
            val intent = Intent(context, DebugHomeActivity::class.java)
            context.startActivity(intent)
        }
    }
}